sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -165,10 +165,18 @@ case class StateStoreMetrics(
 
 object StateStoreMetrics {
   def combine(allMetrics: Seq[StateStoreMetrics]): StateStoreMetrics = {
+    val distinctCustomMetrics = allMetrics.flatMap(_.customMetrics.keys).distinct
+    val customMetrics = allMetrics.flatMap(_.customMetrics)
+    val combinedCustomMetrics = distinctCustomMetrics.map { customMetric =>
```

Comment:
nit but might be necessary: should we only do the sum logic for distinctCustomMetrics only when the overwriting happens? If you agree I'll submit a follow-up PR and let you review. @viirya @HeartSaVioR

Reply:
My understanding is that combine is only called with multiple elements, where we expect these elements will have same custom metrics (as these are from state store provider). Please correct me if I'm missing here.

Reply:
Hm? You mean when there is only one custom metric without duplication? I think doing sum does not get incorrect result. Is there any benefit?

Reply:
So if you assume the chance allMetrics is only having one element, then all these calculations are unnecessary. But this method already assumes the case we want multiple metrics to be combined, and I think we don't blindly call this method with one element. No benefit then.

Reply:
```
I think we don't blindly call this method with one element. No benefit then.
```
Make sense.